statistikk/src/main/kotlin/no/nav/su/se/bakover/statistikk/behandling/søknadsbehandling/BehandlingsstatistikkMapperForSøknadsbehandling.kt

```diff
@@ -0,0 +1,145 @@
+
+        is StatistikkEvent.Behandling.Søknad.Underkjent.Avslag -> toDto(
+            clock = clock,
+            gitCommit = gitCommit,
+            funksjonellTid = this.søknadsbehandling.attesteringer.toFunksjonellTid(this.søknadsbehandling.id, clock),
+            behandlingStatus = BehandlingStatus.Underkjent,
+            behandlingsresultat = BehandlingResultat.AvslagSøknadsbehandling,
+            resultatBegrunnelse = utledAvslagsgrunner(this.søknadsbehandling.avslagsgrunner),
+        )
+
+        is StatistikkEvent.Behandling.Søknad.Iverksatt.Innvilget -> toDto(
+            clock = clock,
+            gitCommit = gitCommit,
+            funksjonellTid = this.vedtak.opprettet,
+            behandlingStatus = BehandlingStatus.Iverksatt,
+            behandlingsresultat = BehandlingResultat.Innvilget,
+            beslutter = this.vedtak.attestant,
+            avsluttet = true,
+        )
+
+        is StatistikkEvent.Behandling.Søknad.Iverksatt.Avslag -> toDto(
+            clock = clock,
+            gitCommit = gitCommit,
+            funksjonellTid = this.vedtak.opprettet,
+            behandlingStatus = BehandlingStatus.Iverksatt,
+            behandlingsresultat = BehandlingResultat.AvslagSøknadsbehandling,
+            resultatBegrunnelse = utledAvslagsgrunner(this.søknadsbehandling.avslagsgrunner),
+            beslutter = this.vedtak.attestant,
+            avsluttet = true,
+        )
+
+        is StatistikkEvent.Behandling.Søknad.Lukket -> {
+            toDto(
+                clock = clock,
+                gitCommit = gitCommit,
+                funksjonellTid = this.søknadsbehandling.lukketTidspunkt,
+                behandlingStatus = BehandlingStatus.Avsluttet,
+                behandlingsresultat = this.søknadsbehandling.toBehandlingResultat(),
+                // Det kreves ikke attestant ved lukking.
+                beslutter = null,
+                totrinnsbehandling = false,
+                avsluttet = true,
+                saksbehandler = this.saksbehandler,
+            )
+        }
+    }
+}
+
+private fun StatistikkEvent.Behandling.Søknad.toDto(
+    clock: Clock,
+    gitCommit: GitCommit?,
+    funksjonellTid: Tidspunkt,
+    behandlingStatus: BehandlingStatus,
+    behandlingsresultat: BehandlingResultat? = null,
+    resultatBegrunnelse: String? = null,
+    beslutter: NavIdentBruker.Attestant? = null,
+    totrinnsbehandling: Boolean = true,
+    avsluttet: Boolean = false,
+    saksbehandler: NavIdentBruker.Saksbehandler? = this.søknadsbehandling.saksbehandler,
+): BehandlingsstatistikkDto {
+    val søknadsbehandling = this.søknadsbehandling
+    val søknad = søknadsbehandling.søknad
+    return BehandlingsstatistikkDto(
+        behandlingType = Behandlingstype.SOKNAD,
+        behandlingTypeBeskrivelse = Behandlingstype.SOKNAD.beskrivelse,
+        funksjonellTid = funksjonellTid,
+        tekniskTid = Tidspunkt.now(clock),
+        registrertDato = søknad.opprettet.toLocalDate(zoneIdOslo),
```

Comment:
Burde dette være behandlingen sin opprettet?

Reply:
>* @param mottattDato Denne datoen forteller fra hvilken dato behandlingen først ble initiert. Datoen brukes i beregning av saksbehandlingstid og skal samsvare med brukerens opplevelse av at saksbehandlingen har startet.
 * @param registrertDato Tidspunkt for når behandlingen ble registrert i saksbehandlingssystemet. Denne kan avvike fra mottattDato hvis det tar tid fra postmottak til registrering i system, eller hvis en oppgave om å opprette behandling ligger på vent et sted i NAV. Ved automatisk registrering av saker er denne samme som mottattDato.
 
Så jeg tenker at mottattDato ved digital søknad er søknad.opprettet, mens ved papirsøknad er den når postmottaket mottok søknaden.
Mens registrertDato er når søknaden ble registrert i vårt system, altså søknad.opprettet (ikke søknadsbehandlingen)

Så det ser rett ut.

Reply:
De er riktige slik som de er. Vi må sende de med første hendelse også er det valgfritt og sende de samme verdiene på nytt i de etterfølgende hendelsene.